pkg/alert/commands.go

```diff
@@ -318,3 +316,187 @@ func NewDeleteAlertSettingConditionCommand() *cobra.Command {
+			} else {
+				alertchannels, err = config.Client.GetAlertChannels(page, pageSize, sortField, sortOrder, templateName, channelName)
+			}
+			if err != nil {
+				golog.Errorf("Failed to retrieve alert channels. [%s]", err.Error())
+				return err
+			}
+
+			if !details {
+				return bite.PrintObject(cmd, alertchannels)
+			}
+			return bite.PrintObject(cmd, alertchannelsWithDetails.Values)
+		},
+	}
+
+	cmd.Flags().IntVar(&page, "page", 1, "--page=1")
+	cmd.Flags().IntVar(&pageSize, "pageSize", 10, "--pageSize=10")
+	cmd.Flags().StringVar(&sortField, "sortField", "", `--sortField="name"`)
+	cmd.Flags().StringVar(&sortOrder, "sortOrder", "", `--sortOrder="asc"`)
+	cmd.Flags().StringVar(&templateName, "templateName", "", `--templateName="test"`)
+	cmd.Flags().StringVar(&channelName, "channelName", "", `--channelName="slack"`)
+	cmd.Flags().BoolVar(&details, "details", false, `--details`)
+
+	bite.CanBeSilent(cmd)
+	bite.CanPrintJSON(cmd)
+
+	cmd.AddCommand(NewDeleteAlertChannelCommand())
+	cmd.AddCommand(NewCreateAlertChannelCommand())
+	cmd.AddCommand(NewUpdateAlertChannelCommand())
+
+	return cmd
+}
+
+//NewDeleteAlertChannelCommand creates `alertchannels delete` command
+func NewDeleteAlertChannelCommand() *cobra.Command {
+	var (
+		channelID string
+	)
+
+	cmd := &cobra.Command{
+		Use:              "delete",
+		Short:            "Delete an alert channel",
+		Example:          `alertchannels delete --channelID="fa0e9b96-1048-4f4c-b776-4e96ca62f37d"`,
+		TraverseChildren: true,
+		SilenceErrors:    true,
+		RunE: func(cmd *cobra.Command, args []string) error {
+			err := config.Client.DeleteAlertChannel(channelID)
+			if err != nil {
+				golog.Errorf("Failed to delete alert channel [%s]. [%s]", channelID, err.Error())
+				return err
+			}
+			return bite.PrintInfo(cmd, "Alert channel [%s] deleted", channelID)
+		},
+	}
+
+	cmd.Flags().StringVar(&channelID, "channelID", "", "Channel ID")
+	cmd.MarkFlagRequired("channelID")
+	bite.CanBeSilent(cmd)
+
+	return cmd
+}
+
+// NewCreateAlertChannelCommand creates `alertchannels create` command
+func NewCreateAlertChannelCommand() *cobra.Command {
+	var (
+		propertiesRaw string
+		channel       = api.AlertChannelPayload{}
+	)
+
+	cmd := &cobra.Command{
+		Use:              "create",
+		Short:            "Create a new alert channel",
+		Example:          `alertchannels create --name="kafka-prd-health" --templateName="Slack" --connectionName="slack-connection" --properties="[{\"key\":\"username\",\"value\":\"@luk\"},{\"key\":\"channel\",\"value\":\"#lenses\"}]"`,
+		SilenceErrors:    true,
+		TraverseChildren: true,
+		RunE: func(cmd *cobra.Command, args []string) error {
+
+			if err := bite.CheckRequiredFlags(cmd, bite.FlagPair{"name": channel.Name, "connectionName": channel.ConnectionName, "templateName": channel.TemplateName, "properties": channel.Properties}); err != nil {
```

Comment:
We can also use `MarkFlagRequired()`

Reply:
As discussed, this is required to check that the necessary flags are populated when using a YAML file.